src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -733,6 +786,51 @@ export default class RadixTransactionBuilder {
         return this
     }
 
+    // system param will be of type AcccountSystem when getState has been implemented everywhere.
+    /*
+    private setState(state: LedgerState, account: RadixAccount) {
+        let newState = {}
+        const address = account.getAddress()
+
+        newState[address] = state ? { ...state[address] } : createInitialState()
+
+        Object.keys(newState[address]).forEach((key) => {
+            account.accountSystems.forEach((system, name) => {
+                // Temporarily only use the systems that have been refactored so far
+                if(['TRANSFER','TOKENS'].includes(system.name)) {
+                    newState[address][key] = newState[address][key] ? newState[address][key] : system.getState()[key]
+                }
+            })
+        })
+        return newState
+    }
+    */
+
+    private addAction(account: RadixAccount, action: Function) {
+        this.accounts.push(account)
+        this.actions.push(action)
+    }
+
+    private executeActions(actions) {
+        actions.reduce((prev, fn) => {
+            return fn(prev)
+        }, this.getInitialState(this.accounts))
+
+        this.actions = []
+    }
+
+    private getInitialState(accounts: RadixAccount[]): LedgerState {
+        let state = {}
+        accounts.forEach(account => {
+            account.accountSystems.forEach((system) => {
```

Comment:
What if you moved the functionality of getting the account state into the account? Seems like the account should now how to compute its state, not the transaction builder

Reply:
Totally agree. Fixed.